libs/owasp-diff-runner/src/main/java/ddf/lib/OwaspDiffRunner.java

```diff
@@ -0,0 +1,184 @@
+
+            request.setGoals(Arrays.asList("dependency-check:check",
+                    "--quiet",
+                    "-pl",
+                    modulesOfChangedPoms,
+                    "-Powasp"));
+            invoker.setMavenHome(new File(mavenHome));
+            System.out.println("-- Maven home: " + mavenHome);
+            System.out.println("-- Local Maven repo: " + localRepo);
+
+            InvocationResult mavenBuildResult;
+
+            try {
+                mavenBuildResult = invoker.execute(request);
+            } catch (MavenInvocationException e) {
+                throw new OwaspDiffRunnerException(
+                        OwaspDiffRunnerException.UNABLE_TO_RUN_MAVEN + modulesOfChangedPoms, e);
+            }
+            if (mavenBuildResult.getExitCode() != 0) {
+                throw new OwaspDiffRunnerException(OwaspDiffRunnerException.FOUND_VULNERABILITIES);
+            }
+        } finally {
+            System.out.println(END_OWASP_AUDIT);
+        }
+
+    }
+
+    private static String getModulesOfChangedPoms() throws OwaspDiffRunnerException {
+        String changedFiles;
+        String currentBranchName;
+
+        try {
+            currentBranchName = IOUtils.toString(runTime.exec(GIT_SHORT_BRANCH_NAME_COMMAND)
+                    .getInputStream())
+                    .replace(File.separator, "")
+                    .replace(System.getProperty("line.separator"), "");
+
+            changedFiles = IOUtils.toString(runTime.exec(
+                    GIT_DIFF_NAME_COMMAND + currentBranchName + "..master")
+                    .getInputStream());
+        } catch (IOException e) {
+            throw new OwaspDiffRunnerException(OwaspDiffRunnerException.UNABLE_TO_RETRIEVE_GIT_INFO,
+                    e);
+        }
+
+        System.out.println(
+                "Comparing commits of branch " + currentBranchName + " to master. Changed poms: ");
+
+        return Arrays.stream(changedFiles.split(System.getProperty("line.separator")))
+                .filter(path -> path.endsWith("pom.xml"))
+                .peek(System.out::println)
+                .map(path -> path.endsWith(File.separator + "pom.xml") ?
+                        path.replace(File.separator + "pom.xml", "") :
+                        path.replace("pom.xml",
+                                File.separator)) //Special case for the root pom, change path pom.xml -> /
+                .collect(Collectors.joining(","));
+    }
+
+    private static String getMavenHome() throws OwaspDiffRunnerException {
+        String mavenHome;
+        String mavenVersionInfo;
+
+        try {
+            mavenVersionInfo = IOUtils.toString(runTime.exec(MAVEN_VERSION_COMMAND)
+                    .getInputStream());
+
+            //parsing console response, confirmed to work with at least maven version 3.3.9
+            mavenHome = Arrays.stream(mavenVersionInfo.split(System.getProperty("line.separator")))
```

Comment:
I wanted to keep it stray away from depending on environments/environment variables. For example, I don't have my M2_HOME set on my own machine. The `mvn -v` is guaranteed to be there though.